CMakeLists.txt

```diff
@@ -0,0 +1,199 @@
+# We need 3.12 or later, so that we can set policy CMP0074; see below.
+cmake_minimum_required(VERSION 3.12)
+
+# We want find_path() and find_library() to honor {packagename}_ROOT
+if(POLICY CMP0074)
+  cmake_policy(SET CMP0074 NEW)
+endif()
+
+# MAIN_PROJECT CHECK
+set(PCAPPP_MAIN_PROJECT OFF)
+if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(PCAPPP_MAIN_PROJECT ON)
+endif()
+
+# Include git module required for version
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/")
+include(VersionFromGit)
+
+# Get the git version
+version_from_git(
+  LOG
+  ON
+  TIMESTAMP
+  "%Y%m%d%H%M%S")
```

Comment:
I'm not sure exactly how this function works, but my understanding is that the version will be `22.05` even if it's not an official version, is that correct?
For example - if I build this PR, the version will be `22.05` although it contains a lot of code that wasn't included in the `22.05` release. I'd expect the version for unofficial versions to be `22.05+` or something else that shows it's not an official version

Reply:
At the moment I didn't update the PcapPlusPLusversion.h to keep compatibility with the previous build system.

But if we move to git we can use the semver version without the date and give an output 
version 22.05 when building on a tag 
version 22.05+gc41aac00 when build dev release

Reply:
but who will update `PcapPlusPlusVersion.h` in each commit? 🤔

Reply:
For the moment we still need to handle it manually but as soon as we drop the previous build system it can be be automatic.

Reply:
I'm curious how we can automate it? The CI will make a commit to update the version?
Another option is to create `PcapPlusPlusVersion.h` as part of the CMake build process, but it seem way to complex for what we need. I still think we can keep the current system where all non-official versions have only one version number (like `v22.05+`)

Reply:
The trig is still manual, the action is a Maintainer create a tag and push it.
The push of the tag then trig a build that will be the official one. And later it should also trig the release process and upload the artifact.

It's really easy to generate a PcapPlusPlusVersion.h with CMake or we can pass the version as Define both are quite simple. The only issue I see is keeping the legacy build system.

At the moment the CMake only use the version for:
- .so install that will create libpcap++.so which is a symlink to libpcap++.so.22.05 for example.
- .pc the version set in the PkgConfig file
- .cmake the version set in the CMake version file

Reply:
should be good now.

Reply:
I think it works now, thanks!

However, the fact that we automatically pull the tags from git and then add `+` if it's a non-official version adds complexity to the CI because we need to pull all the git history. What if we do something a bit more manual but far more simple:
```CMake
# In CMakeLists.txt
set(PCAPPP_VERSION "22.05+")

# In Common++/CMakeLists.txt 
set_property(TARGET Common++ PROPERTY VERSION ${PCAPPP_VERSION})

# Do the same in Packet++/CMakeLists.txt Pcap++CMakeLists.txt or wherever we currently use ${SEMVER}

# In cmake/PcapPlusPlus.pc.in
Version: @PCAPPP_VERSION@
```

When we want to bump up a version (which currently happens every 6 months or so) we change `PCAPPP_VERSION` and also the version in `PcapPlusPlusVersion.h`

I know it seems a bit manual, but it removes a lot of complexity because we don't need `cmake/VersionFromGit.cmake` anymore and we don't need these extra additions in GitHub Actions and CirrusCI which are meant to pull the git history.

Please let me know what you think.

My thinking is that we can start simple and add more automatic processes later if/when we need to.

Reply:
@seladb the Project Version doesn't like the '+' in "22.05+" see 
https://cmake.org/cmake/help/latest/variable/CMAKE_PROJECT_VERSION.html

Reply:
@clementperon where do you see this error? Can you share a link?
What is the recommended way to mark non-official versions?

Reply:
I have pushed the commit that make the Project version failed.

At the moment there is not support for + or -DEV suffit but maybe 22.05.999 will do the job

Reply:
I'm a bit confused 🤔 
Previously you suggested dev versions that look like this: `22.05+gc41aac00` https://github.com/seladb/PcapPlusPlus/pull/944#discussion_r969531788

But if `+` isn't supported - how did it work?

Reply:
The project version was not properly set but maybe if we overwrite everywhere it's used it will work fine.

Reply:
It looks like it works fine now!